build.sh

```diff
@@ -83,27 +88,28 @@ shift $(( OPTIND - 1 ));
 
 # Sanity checks for required arguments
 if [ -z "${GIT_BRANCH}" ]; then
-  echo "Error: Git branch with -b is not set."
-  usage
-  exit 1
+  GIT_BRANCH=`git status | grep "On branch" | sed -r "s/On branch //g" | sed -r "s/\//-/g"`
```

Comment:
You should use ```git symbolic-ref --short 'HEAD' ```.

There is a problem with branch detection. Pleas also consider the following cases:
- You are in detached HEAD mode (check out by commit id)
- You have multiple branches pointing to this commit

In jeninks, this gets even worse. Jenkins scans for all branches and selects the ones with the latest commits.

I whould like to use ```snapshot``` as branch name for that case (and don't use a build number).